chroot_setup.py

```diff
@@ -0,0 +1,373 @@
+def chroot_cp ( source: str ):
+  print("cp $1 $CHROOT_DIR$1")
+  if (CHROOT_DEBUG == False):
+    shutil.copy(source, args.chroot_directory + source)
+
+# TODO: documentation
+def chroot_cp_dir ( source: str ):
+  print("cp -R $1/* $CHROOT_DIR$1/.")
+  # TODO: Why mkdir before checking debug?
+  chroot_mkdir(source)
+  if (CHROOT_DEBUG == False):
+    shutil.copytree(source, args.chroot_directory + source)
+
+
+def chroot_chmod (mode, file)
+  # TODO: change 
+  printf("chroot $CHROOT_DIR /QOpenSys/usr/bin/bsh -c \"chmod $1 $2\"")
+  if (CHROOT_DEBUG == False):
+    os.chmod(args.chroot_directory + file, int(mode), 8)
+
+def chroot_chmod_dir (mode, path) {
+  for root, dirs, files in os.walk(args.chroot_directory + path):  
+    for dir in dirs:  
+      os.chmod(os.path.join(root, dir), int(mode, 8))
+    for file in files:
+      os.chmod(os.path.join(root, file), int(mode, 8))
+
+def chroot_chown (user: str, file: str)
+  # TODO: change 
+  printf("chroot $CHROOT_DIR /QOpenSys/usr/bin/bsh -c \"chmod $1 $2\"")
+  if (CHROOT_DEBUG == False):
+    os.chown(args.chroot_directory + file, user)
+
+#TODO: documentation
+def chroot_chown_dir (user: str , path: str) {
+  for root, dirs, files in os.walk(args.chroot_directory + path):  
+    for dir in dirs:  
+      os.chown(os.path.join(root, dir), user)
+    for file in files:
+      os.chown(os.path.join(root, file), user)
+
+# TODO: documentation
+def chroot_system(line: str):
+  line = line.rstrip()
+  print('system -i "' + line + '"')
+  if (CHROOT_DEBUG == False):
+    os.system('system -i "' + line + '"')
+
+# TODO: documentation
+def chroot_sh(line: str):
+  print(line)
+  if (CHROOT_DEBUG == False):
+    os.system(line)
+
+action_dictionary = {
+  ':file'       : chroot_setup,
+  ':mkidr'      : chroot_mkdir,      # done
+  ':ln_fix_rel' : chroot_ln_fix_rel, # done
+  ':ln_rel'     : chroot_ln_rel,     # done
+  ':ln'         : chroot_ln,         # done
+  ':mknod'      : chroot_mknod,      # done
+  ':cp_dir'     : chroot_cp_dir,     # done
+  ':cp'         : chroot_cp,         # done
+  ':chmod_dir'  : chroot_chmod_dir,  # done
+  ':chmod'      : chroot_chmod,      # done
+  ':chown_dir'  : chroot_chown_dir,  # done
+  ':chown'      : chroot_chown,      # done
+  ':system'     : chroot_system,
+  ':sh'         : chroot_sh
+}
+
+def chroot_setup (chroot_lst):
+  lst_file = open(chroot_lst, 'r')
+  action = None
+
+  # Iterate over all lines in the file, executing actions where encountered
+  for line in lst_file:
+    line = line.strip()
+
+    # Blank line
+    if not line:
+      # print('empty')
+
+    # Comment
+    else if line.startswith('#'):
+      # print('comment')
+
+    # Determine action
+    else if line.startswith(':'):
+      action = action_dictionary[line]
+
+    # Execute action
+    else:
+      if action not None:
+        action(line)
+
+# TODO: documentation
+def initial_check ():
+  if (os.path.exists('/QOpenSys/usr/bin')):
+    CHROOT_DEBUG = False
```

Comment:
By Python conventions this would be a constant, but it's being altered below. Instead, we can drop this variable and reference args.verbose instead.

In addition, @abmusse mentioned using args.verbose to adjust the logging level and then just calling the appropriate logging function and we never have to even check the variable anywhere else.

Reply:
I guess this _is_ separate from verbosity of logging as when this is set, no action will be performed but it will still go through showing what would be done. Many utilities have command line options for this, usually with `--dry-run` or `-n`. I think that would make more sense to do something like that.

Reply:
To me the `CHROOT_DEBUG` variable was a misnomer because it really used to guard against running commands on non IBM i sytems as it was only set to true in the else leg of `initial_check` when the script was run on non IBM i system then the script exited. 

Therefore none of the actions would run anyway so why did the script even bother to include checks for `CHROOT_DEBUG` in the actions.

Reply:
I think we can get rid of all the `if not CHROOT_DEBUG` in the chroot_* actions. We just need to exit the script properly in the inital_check if we are not running on IBM i.

Reply:
Fixed up in 1991e26